appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockAreaChartDataModel.java

```diff
@@ -0,0 +1,28 @@
+package com.google.appinventor.client.editor.simple.components;
+
+import org.pepstock.charba.client.data.Data;
+
+public class MockAreaChartDataModel extends MockLineChartBaseDataModel {
+    public MockAreaChartDataModel(Data chartData) {
+        super(chartData);
+    }
+
+    @Override
+    protected void setDefaultStylingProperties() {
+        super.setDefaultStylingProperties();
+        dataSeries.setFill(true);
+    }
+
+    @Override
+    protected void addDataSeriesToChart() {
+        if (chartData.getDatasets().size() == 0) {
+            chartData.setDatasets(dataSeries);
+        } else {
+            // Area Chart draw order is reversed in MPAndroidChart,
+            // so we have to reverse the order in which we add
```

Comment:
Look into omitting this, inserting in reverse order for the Android implementation, and then reversing the order of the Legend.